packages/core/test/tabs/tabs2Tests.tsx

```diff
@@ -0,0 +1,274 @@
+        });
+    });
+
+    it("clicking nested tab should not affect parent", () => {
+        const changeSpy = sinon.spy();
+        const wrapper = mount(
+            <Tabs2 id={ID} onChange={changeSpy}>
+                {getTabsContents()}
+                <Tab2 id="nested" title="Nested">
+                    <Tab2 id="last" title="Click me" />
+                </Tab2>
+            </Tabs2>,
+            { attachTo: testsContainerElement },
+        );
+        assert.equal(wrapper.state("selectedTabId"), TAB_IDS[0]);
+        // last Tab is inside nested
+        wrapper.find(Tab2).last().simulate("click");
+        assert.equal(wrapper.state("selectedTabId"), TAB_IDS[0]);
+        assert.isTrue(changeSpy.notCalled, "onChange invoked");
+    });
+
+    it("changes tab focus when arrow keys are pressed", () => {
+        const wrapper = mount(
+            <Tabs2 id={ID}>
+                <Tab2 id="first" title="First"><strong>first panel</strong></Tab2>,
+                <Tab2 disabled id="second" title="Second"><strong>second panel</strong></Tab2>,
+                <Tab2 id="third" title="Third"><strong>third panel</strong></Tab2>,
+            </Tabs2>,
+            { attachTo: testsContainerElement },
+        );
+
+        const tabList = wrapper.find(TAB_LIST);
+        const tabElements = testsContainerElement.queryAll(TAB);
+        (tabElements[0] as HTMLElement).focus();
+
+        tabList.simulate("keydown", { which: Keys.ARROW_RIGHT });
+        assert.equal(document.activeElement, tabElements[2], "move right and skip disabled");
+        tabList.simulate("keydown", { which: Keys.ARROW_RIGHT });
+        assert.equal(document.activeElement, tabElements[0], "wrap around to first tab");
+        tabList.simulate("keydown", { which: Keys.ARROW_LEFT });
+        assert.equal(document.activeElement, tabElements[2], "wrap around to last tab");
+        tabList.simulate("keydown", { which: Keys.ARROW_LEFT });
+        assert.equal(document.activeElement, tabElements[0], "move left and skip disabled");
+    });
+
+    it("enter and space keys click focused tab", () => {
+        const changeSpy = sinon.spy();
+        const wrapper = mount(
+            <Tabs2 id={ID} onChange={changeSpy}>{getTabsContents()}</Tabs2>,
+            { attachTo: testsContainerElement },
+        );
+        const tabList = wrapper.find(TAB_LIST);
+        const tabElements = testsContainerElement.queryAll(TAB);
+
+        // must target different elements each time as onChange is only called when index changes
+        tabList.simulate("keypress", { target: tabElements[1], which: Keys.ENTER });
+        tabList.simulate("keypress", { target: tabElements[2], which: Keys.SPACE });
+
+        assert.equal(changeSpy.callCount, 2);
+        assert.deepEqual(changeSpy.args, [
+            [TAB_IDS[1], TAB_IDS[0]],
+            [TAB_IDS[2], TAB_IDS[1]],
+        ]);
+    });
+
+    it("animate=false removes moving indicator element", () => {
```

Comment:
Yeah I feel like we should keep the blue underline on the selected tab, even if it doesn't animate. Was there a design discussion in the past about this point?